### PR TITLE
Bug/edit mode doble link

### LINKED ISSUE #1070: Double Link to an existing board shows in editing mode
When choosing a folder to edit through Edit option, two dropdown menus 'Link to an existing board' show.
When creating new folder, it is ok - only one dropdown.

Capture2

### LINKED ISSUE #1088: Edit tile - Link an existing board dont work properly
When choosing a folder to edit through Edit option,  the dropdown menu 'Link to an existing board' dont show the linked board. 
image

### PR DESCRIPTION
**Pull request by RodriSanchez1**
close #1088 and  close #1070

### CODE REVIEW

File: src/components/Board/TileEditor/TileEditor.component.js [hunk 1]
```diff
@@ -552,11 +561,6 @@ export class TileEditor extends Component {
                       onChange={this.handleVocalizationChange}
                       fullWidth
                     />
-                    <div>
-                      {this.editingTile() &&
-                        tileInView.loadBoard &&
-                        selectBoardElement}
-                    </div>
```
Comment (RodriSanchez1) [inline src/components/Board/TileEditor/TileEditor.component.js:559]: here solves #1070

File: src/components/Board/Board.container.js [hunk 2]
```diff
@@ -1614,7 +1614,12 @@ export class BoardContainer extends Component {
           onClose={this.handleTileEditorCancel}
           onEditSubmit={this.handleEditTileEditorSubmit}
           onAddSubmit={this.handleAddTileEditorSubmit}
-          boards={this.props.boards}
+          boards={this.props.boards.filter(
+            board =>
+              board !== null &&
+              board.id !== null &&
+              this.props.communicator.boards.includes(board.id)
+          )}
```
Comment (RodriSanchez1) [inline src/components/Board/Board.container.js:1622]: filter boards props to get only boards of the current user loged in

File: src/components/Board/Board.container.js [hunk 3]
```diff
@@ -1614,7 +1614,12 @@ export class BoardContainer extends Component {
           onClose={this.handleTileEditorCancel}
           onEditSubmit={this.handleEditTileEditorSubmit}
           onAddSubmit={this.handleAddTileEditorSubmit}
-          boards={this.props.boards}
+          boards={this.props.boards.filter(
```
Comment (martinbedouret) [inline src/components/Board/Board.container.js:1617]: this Looks incorrect to me. The boards included in the default cboard are not owned by the user who logged in.  You should not filter boards.

File: src/components/Board/TileEditor/TileEditor.component.js [hunk 4]
```diff
@@ -414,12 +412,19 @@ export class TileEditor extends Component {
     this.updateTileProperty('image', image);
   };
 
+  linkedBoard = () => {
```
Comment (martinbedouret) [inline src/components/Board/TileEditor/TileEditor.component.js]: to me, the linked board looks an internal state of the component.

**Review by martinbedouret — CHANGES_REQUESTED** [review]

File: src/components/Board/Board.container.js (hunk 3 above)
Comment (RodriSanchez1) [reply src/components/Board/Board.container.js:1617]: If you dont filter, the select will show all the boards  in boards state. That contains the boards  created  before the user log in  and the boards created after the user log in. The filter takes only the user boards.
image
In the image you can see two home boards. You can check this issue, its a similar problem  #1102

**Review by martinbedouret — APPROVED** [review]

**Pull request merged**